main.py

```diff
@@ -110,7 +110,12 @@ async def on_message(message: Message):
             await message.channel.send("We're busy busting.")
             return
 
-        await command_list(message)
+        thumbs_up = "\N{THUMBS UP SIGN}"
+        await message.add_reaction(thumbs_up)
+
+        # Ensure two scrapes aren't making/deleting files at the same time
+        async with list_task_control_lock:
```

Comment:
There is not, ideally it should be cancelled. I was a bit wary of doing so in the first draft just because of the hassle with async cancelling in the "Pin now playing for duration of song" commit and also having to deal with the partial state a cancel could leave global variables in, so I thought that should be just a separate additional commit since I want cache functionality in quick.

Reply:
Sorry, I meant cancelling the incoming request to list until the existing one had completed, which would be much more straightforward than cancelling an existing list.

Reply:
Oh yeah, that's a good idea. I'll just implement a different react to say the !list has been ignored

Reply:
Yeah, either that or just a message saying "A !list is already in progress" would work.